src/test/rpc/AccountObjects_test.cpp

```diff
@@ -316,10 +316,225 @@ class AccountObjects_test : public beast::unit_test::suite
         }
     }
 
+    void testObjectTypes()
+    {
+        testcase ("object types");
+
+        // Give gw a bunch of ledger objects and make sure we can retrieve
+        // them by type.
+        using namespace jtx;
+
+        Account const alice { "alice" };
+        Account const gw{ "gateway" };
+        auto const USD = gw["USD"];
+
+        // Test for ticket account objects when they are supported.
+        Env env(*this, supported_amendments().set(featureTickets));
+
+        // Make a lambda we can use to get "account_objects" easily.
+        auto acct_objs = [&env] (Account const& acct, char const* type)
+        {
+            Json::Value params;
+            params[jss::account] = acct.human();
+            params[jss::type] = type;
+            params[jss::ledger_index] = "validated";
+            return env.rpc("json", "account_objects", to_string(params));
+        };
+
+        // Make a lambda that easily identifies empty account objects.
+        auto acct_objs_is_empty =
+            [&env, &acct_objs] (Account const& acct, char const* type)
+        {
+            Json::Value const resp = acct_objs (acct, type);
+
+            bool const isArray = resp["result"]["account_objects"].isArray();
+            return isArray && (resp["result"]["account_objects"].size() == 0);
+        };
```

Comment:
Suggestion: If the lambda takes a size param, it can be called from other instances in the test.

```
        auto acct_objs_is_size =
            [&env, &acct_objs](Account const& acct, char const* type, unsigned size)
        {
            Json::Value const resp = acct_objs(acct, type);

            return resp["result"]["account_objects"].isArray() &&
                resp["result"]["account_objects"].size() == size;
        };
```
These instances 
```
            BEAST_EXPECT (resp["result"]["account_objects"].isArray());
            BEAST_EXPECT (resp["result"]["account_objects"].size() == 1);
```
Can become
```
          BEAST_EXPECT (acct_objs_is_size(gw, jss::state, 1));
```